In [12]:
import numpy as np
import pandas as pd
import statistics
brady_bunch = [6,8,8,10,11,12,14]

In [17]:
def summary_stats(nums):
    print('mean: ',np.mean(nums))
    print('median: ',(np.median(nums)))
    try:
        print('mode: ',statistics.mode(nums))
    except Exception:
        print('mode: no unique mode found')
    print('variance: ',np.var(nums))
    print('Standard Deviantion: ',np.std(nums, ddof=1))
    print('Standard Error: ',np.std(nums, ddof=1)/np.sqrt(len(nums)))

In [14]:
summary_stats(brady_bunch)

mean:  9.857142857142858
median:  10.0
mode:  8
variance:  6.408163265306122
Standard Deviantion:  2.734262327610589
Standard Error:  1.0334540197243192


## Central Tendency
The median in a better measure of central tendency because a majority of malues are closer or equal to the median vs the mean

## Variance
Standard Deviantion is a better measure of variance because the upper and lower bounds of the dataset are least impacted by 2 standard deviations of 2.734 from the mean vs a variance of 6.408 from the mean.

In [15]:
# Cindy has a birthday
brady_bunch = [7,8,8,10,11,12,14]
summary_stats(brady_bunch)

mean:  10.0
median:  10.0
mode:  8
variance:  5.428571428571429
Standard Deviantion:  2.516611478423583
Standard Error:  0.9511897312113418


## Observations
The mean increased is now identical to the median
all measures of variance decreased

In [27]:
# No one like Cousin Oliver
brady_bunch = [7,8,1,10,11,12,14]
summary_stats(brady_bunch)

mean:  9.0
median:  10.0
mode: no unique mode found
variance:  15.428571428571429
Standard Deviantion:  4.242640687119285
Standard Error:  1.6035674514745462


## observations
mean skewed down due to outlier


In [35]:
# Brady bunch survey
survey = [.2,.23,.17,.05]
# SciPhi Phanatic 5% irrevelant
survey.pop()

0.05

In [39]:
#get average of remaining surveys for population estimate
print(np.mean(survey)*100,"%")

20.000000000000004 %
